src/components/screens/SendScreen/screens/TransactionValueScreen.tsx

```diff
@@ -0,0 +1,286 @@
+import { BalanceRow } from "components/BalanceRow";
+import BottomSheet from "components/BottomSheet";
+import ContextMenuButton from "components/ContextMenuButton";
+import { BaseLayout } from "components/layout/BaseLayout";
+import { 
+  ContactRow, 
+  NumericKeyboard, 
+  TransactionReviewBottomSheet 
+} from "components/screens/SendScreen/components";
+import { TransactionProcessingScreen } from "components/screens/SendScreen/screens";
+import { Button } from "components/sds/Button";
+import Icon from "components/sds/Icon";
+import { Display, Text } from "components/sds/Typography";
+import { SEND_PAYMENT_ROUTES, SendPaymentStackParamList } from "config/routes";
+import { useAuthenticationStore } from "ducks/auth";
+import { formatAssetAmount, formatFiatAmount } from "helpers/formatAmount";
+import useAppTranslation from "hooks/useAppTranslation";
+import { useBalancesList } from "hooks/useBalancesList";
+import useColors from "hooks/useColors";
+import useGetActiveAccount from "hooks/useGetActiveAccount";
+import { useTokenFiatConverter } from "hooks/useTokenFiatConverter";
+import React, { useEffect, useMemo, useRef, useState } from "react";
+import { TouchableOpacity, View } from "react-native";
+
+type TransactionValueScreenProps = NativeStackScreenProps<
+  SendPaymentStackParamList,
+  typeof SEND_PAYMENT_ROUTES.TRANSACTION_VALUE_SCREEN
+>;
+
+/**
+ * TransactionValueScreen Component
+ *
+ * A screen for entering transaction values in either token or fiat currency.
+ * Supports switching between token and fiat input modes with automatic conversion.
+ *
+ * @param {TransactionValueScreenProps} props - Component props
+ * @returns {JSX.Element} The rendered component
+ */
+const TransactionValueScreen: React.FC<TransactionValueScreenProps> = ({
+  navigation,
+  route,
+}) => {
+  const { t } = useAppTranslation();
+  const { themeColors } = useColors();
+  const { address, tokenId } = route.params;
+  const { account } = useGetActiveAccount();
+  const { network } = useAuthenticationStore();
+  const publicKey = account?.publicKey;
+  const reviewBottomSheetModalRef = useRef<BottomSheetModal>(null);
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  const { balanceItems } = useBalancesList({
+    publicKey: publicKey ?? "",
+    network,
+    shouldPoll: false,
+  });
+
+  const selectedBalance = balanceItems.find((item) => item.id === tokenId);
+
+  const {
+    tokenValue,
+    fiatValue,
+    showDollarValue,
+    setShowDollarValue,
+    handleValueChange,
+    handlePercentagePress,
+  } = useTokenFiatConverter({ selectedBalance });
+
+  const menuActions = useMemo(
+    () => [
+      {
+        title: t("transactionValueScreen.menu.fee", { fee: "0.025" }),
+        systemIcon: "arrow.trianglehead.swap",
+        onPress: () => {
+          navigation.navigate(SEND_PAYMENT_ROUTES.TRANSACTION_FEE_SCREEN, {
+            address,
+            tokenId,
+            tokenCode: selectedBalance?.tokenCode ?? "XLM",
```

Comment:
I don't think we need to pass any of those params here as fees are always charged in XLM